src/plugins/intel_gpu/include/intel_gpu/runtime/layout.hpp

```diff
@@ -126,58 +126,107 @@ inline data_types element_type_to_data_type(ov::element::Type t) {
+        const auto cnt = std::min(other.size(), SHAPE_RANK_MAX);       
+        std::copy_n(std::begin(other), cnt, _dynamic_pad_dims);
+        std::fill_n(_dynamic_pad_dims + cnt, SHAPE_RANK_MAX - cnt, 0);
     }
 
-    tensor get_dynamic_pad_dims() const {
-        return _dynamic_pad_dims;
+    std::vector<tensor::value_type> get_dynamic_pad_dims(const size_t max_size = 0) const {
+        std::vector<tensor::value_type> sizes(max_size == 0 ? sizeof(_dynamic_pad_dims) / sizeof(_dynamic_pad_dims[0]) : max_size, 0);
+        for (size_t i = 0; i < sizes.size(); ++i)
+            sizes[i] = _dynamic_pad_dims[i];
+        return sizes;
     }
+
     /// @brief
     /// @param lower_sizes Top-left padding sizes. See @ref tensor::tensor(const std::vector<value_type>&, value_type) for details.
     /// @param upper_sizes Bottom-right padding sizes. See @ref tensor::tensor(const std::vector<value_type>&, value_type) for details.
     /// @param filling_value Filling value for padding area.
     padding(const std::vector<tensor::value_type>& lower_sizes,
             const std::vector<tensor::value_type>& upper_sizes,
             float filling_value = 0.0f,
-            const tensor& dynamic_pad_dims = tensor(0))
-        : _lower_size(to_abs(lower_sizes), 0),
-          _upper_size(to_abs(upper_sizes), 0),
-          _filling_value(filling_value),
-          _dynamic_pad_dims(dynamic_pad_dims) {}
+            const std::vector<tensor::value_type>& dynamic_pad_dims = {})
+        : _filling_value(filling_value) {
+            // paddings
+            OPENVINO_ASSERT(lower_sizes.size() <= SHAPE_RANK_MAX);
+            OPENVINO_ASSERT(upper_sizes.size() <= SHAPE_RANK_MAX);
+            OPENVINO_ASSERT(dynamic_pad_dims.size() <= SHAPE_RANK_MAX);
+            if(lower_sizes.size() > 0) std::copy_n(to_abs(lower_sizes).begin(), lower_sizes.size(), _lower_size);
+            if(upper_sizes.size() > 0) std::copy_n(to_abs(upper_sizes).begin(), upper_sizes.size(), _upper_size);
+            if(dynamic_pad_dims.size() > 0) std::copy_n(to_abs(dynamic_pad_dims).begin(), dynamic_pad_dims.size(), _dynamic_pad_dims);
+          }
 
     /// @brief Constrcuts symmetric padding.
     /// @param sizes Top-left and bottom-right padding sizes. See @ref tensor::tensor(const std::vector<value_type>&,
     /// value_type) for details.
     /// @param filling_value Filling value for padding area.
-    explicit padding(const std::vector<tensor::value_type>& sizes, float filling_value = 0.0f, const tensor& dynamic_pad_dims = tensor(0))
+    explicit padding(const std::vector<tensor::value_type>& sizes, float filling_value = 0.0f, const std::vector<tensor::value_type>& dynamic_pad_dims = {})
         : padding(sizes, sizes, filling_value, dynamic_pad_dims) {}
 
     /// @brief Constructs "zero-sized" padding.
-    padding() : padding({0, 0, 0, 0}, 0, tensor(0)) {}
+    padding() : padding({}, 0, {}) {}
+
+    /// @brief Copy construction.
+    padding(const padding& other) {
```

Comment:
Does the `padding` class still make sense? Maybe we can just have these padding arrays as a part of `layout` ? Won't it help to avoid extra copies?

Reply:
Yes, it still make sense, unless more effort to rewrite the following methods -
 
1. padding class is used by so many internal functions as return value or input parameter. for example, it is used to construct layout. 

If we remove it, these functions would look tedious, and also lack of extensibility.

2. padding class is also directly accessed by functions and tests internally and extensively.  

If we remove the class, we have to re-examine each of them.